### PR TITLE
/reminders: Support many recipients

### PR DESCRIPTION
**Pull request by julienw**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -109,7 +109,7 @@
     "arrow-body-style": 2,
     "arrow-spacing": 2,
     "constructor-super": 2,
-    "no-confusing-arrow": 2,
+    "no-confusing-arrow": [2, { "allowParens": true }],
```
Comment (julienw) [inline .eslintrc.json:112]: see below

File: dao/reminders.js [hunk 2]
```diff
@@ -97,7 +115,11 @@ module.exports = {
         'SELECT * FROM reminders WHERE family = ? AND id = ?',
         family, id
       ))
-      .then(row => row || Promise.reject(notFoundError(id)));
+      .then(reminder => (
+        reminder
+          ? deserialize(reminder)
+          : Promise.reject(notFoundError(id))
+      ));
```
Comment (julienw) [inline dao/reminders.js:122]: previous eslintrc change is for this to not be warned :)

File: utils/object_validator.js [hunk 3]
```diff
@@ -7,5 +7,19 @@ module.exports = {
...
+      throw new InvalidInputError(
+        'invalid_type', `"${prop}" should be an array`
+      );
+    }
+
+    if (obj[prop].length < minLength) {
+      throw new InvalidInputError(
+        'invalid_type', `"${prop}" should have at least ${minLength} elements`
+      );
+    }
```
Comment (julienw) [inline utils/object_validator.js:23]: I decided to do an adhoc check function for arrays

File: dao/reminders.js [hunk 4]
```diff
@@ -112,14 +134,19 @@ module.exports = {
 
   update(family, id, updatedReminder) {
     debug('update(family=%s, id=%s)', family, id);
+
+    checkIsArray(updatedReminder, 'recipients', 1);
+    checkPropertyType(updatedReminder, 'action', 'string');
+    checkPropertyType(updatedReminder, 'due', 'number');
```
Comment (julienw) [inline dao/reminders.js:140]: we were not checking anything for update...

File: dao/reminders.js [hunk 5]
```diff
@@ -28,6 +30,19 @@ function checkUpdateDelete(mode, id) {
   };
 }
 
+function serializeRecipients(recipients) {
+  return JSON.stringify(recipients);
```
Comment (julienw) [inline dao/reminders.js:34]: as we agreed

**Comment by julienw** [comment]
r? @JohanLorenzo 
I rebased your first commit on top of master.
I added a second commit to fix what I thought needed some fixes :)

tell me what you think !

File: dao/reminders.js (hunk 2 above)
Comment (JohanLorenzo) [inline dao/reminders.js:122]: Noted. I got that error too, but I was unsure what were the potential harmful effects

File: utils/object_validator.js (hunk 3 above)
Comment (JohanLorenzo) [inline utils/object_validator.js:23]: 👍

**Comment by JohanLorenzo** [comment]
r=me! 

Closes #19

**Pull request merged**